src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/DefinePropertiesOnce.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Xml.Linq;
+
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class DefinePropertiesOnce : MsBuildProjectFileAnalyzer
+{
+    public DefinePropertiesOnce() : base(Rule.DefinePropertiesOnce) { }
+
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        var props = new HashSet<Node>(new PropertyComparer());
+
+        foreach (var prop in context.Project.PropertyGroups.SelectMany(g => g.Children()))
+        {
+            if (!props.Add(prop))
+            {
+                context.ReportDiagnostic(Descriptor, prop, prop.LocalName);
+            }
+        }
+    }
+
+    private sealed class PropertyComparer : IEqualityComparer<Node>
+    {
+        public bool Equals(Node x, Node y)
+            => Name(x) == Name(y)
```

Comment:
Are you sure that we there are no nodes that are expected to be defined multiple times? Or will we introduce a whitelist when we run into them in the widl?

Reply:
as far as I know, those do not exist. So indeed, if it turns out they exist, we can add an ingore list.